src/Server/modules/WorldPlaceManager.js

```diff
@@ -0,0 +1,401 @@
+						console.log('DELETED DEATH');
+						result = true;
+						break;
+					}
+				}
+			}
+		}
+	}
+	else if (tod == 3) {
+		for (var i = 0; i < allWorldPlace.length; ++i) {
+			if (country == allWorldPlace[i].country && province == allWorldPlace[i].state) {
+				for (var key in allWorldPlace[i].numRecovered) {
+					var temp = new Date(key);
+					var d =
+						temp.getMonth() +
+						1 +
+						'/' +
+						temp.getDate() +
+						'/' +
+						temp.getFullYear();
+					if (d == date) {
+						delete allWorldPlace[i].numRecovered[key];
+						allWorldPlace[i].currentNumRecovered = analytics.GetMostRecentValue(allWorldPlace[i].numRecovered);
+						console.log('DELETED RECOVERED');
+						result = true;
+						break;
+					}
+				}
+			}
+		}
+	}
+	console.log("After:\n", allWorldPlace[11]);
+	return result;
+}
+
+function GetTwoPlacesComparison(country1, province1, country2, province2, tod) {
+    var province1Data = null;
+    var province2Data = null;
+
+	for (var i = 0; i < allWorldPlace.length; ++i) {
+		if (allWorldPlace[i].country == country1 && allWorldPlace[i].state == province1) {
+			province1Data = allWorldPlace[i];
+		}
+		if (allWorldPlace[i].country == country2 && allWorldPlace[i].state == province2) {
+			province2Data = allWorldPlace[i];
+		}
+	}
+
+    if (province1Data == null || province2Data == null) {
+        place1Value = (province1Data == null) ? -1 : 0;
+		place2Value = (province2Data == null) ? -1 : 0;
+        return { place1Value: place1Value, place2Value: place2Value }
+	}
+	
+	var place1Value, place2Value;
+
+    switch (tod) {
+		case '1': // confirmed
+			console.log("in case");
+            place1Value = province1Data.currentNumConfirmed;
+            place2Value = province2Data.currentNumConfirmed;
+            break;
+        case '2': // dead
+            place1Value = province1Data.currentNumDeaths;
+            place2Value = province2Data.currentNumDeaths;
+            break;
+        case '3': // recovered
+            place1Value = province1Data.currentNumRecovered;
+            place2Value = province2Data.currentNumRecovered;
+            break;
+	}
+	
+    return { place1Value: place1Value, place2Value: place2Value }
+}
+
+function GetWorldPopulationAnalysis(country, province) {
+    var recovered, deaths, confirmed;
+
+    var provinceData = null;
+
+    for (var i = 0; i < allWorldPlace.length; ++i) {
+        if (allWorldPlace[i].country == country && allWorldPlace[i].state == province) {
+            provinceData = allWorldPlace[i];
+        }
+    }
+
+    if (provinceData == null) { // The requested province does not exist.
+        return { recovered: -1, deaths: -1, confirmed: -1 }
+    }
+
+    deaths = provinceData.currentNumDeaths;
+    confirmed = provinceData.currentNumConfirmed;
+	recovered = provinceData.currentNumRecovered;
+
+    return { recovered: recovered, deaths: deaths, confirmed: confirmed }
+}
+
+function GetRows(country, province, tod) {
```

Comment:
Check if allWorldPlace is undefined at the beginning of the function.

Reply:
added